libibverbs/ibdev_nl.c

```diff
@@ -217,3 +217,41 @@ int find_sysfs_devs_nl(struct list_head *tmp_sysfs_dev_list)
 	nl_socket_free(nl);
 	return EINVAL;
 }
+
+static int get_copy_on_fork_cb(struct nl_msg *msg, void *data)
+{
+	struct nlattr *tb[RDMA_NLDEV_ATTR_MAX];
+	int ret;
+
+	ret = nlmsg_parse(nlmsg_hdr(msg), 0, tb, RDMA_NLDEV_ATTR_MAX - 1,
+			  rdmanl_policy);
+	if (ret < 0)
+		return ret;
+
+	/* Older kernels don't support COF and don't report it through nl */
+	if (!tb[RDMA_NLDEV_SYS_ATTR_COPY_ON_FORK]) {
+		*(uint8_t *)data = 0;
+		return NL_OK;
+	}
+
+	*(uint8_t *)data = nla_get_u8(tb[RDMA_NLDEV_SYS_ATTR_COPY_ON_FORK]);
+	return NL_OK;
+}
+
+bool get_copy_on_fork(void)
```

Comment:
Do you except it to change during the RDMA device's lifetime (kernel patch always reports it as supported)?
Seems kind of wasteful to create a socket -> talk with the kernel -> close socket
on every call to ibv_is_fork_initialized() if the result never changes.

Maybe store it (like we do with ibdev_idx and such) while in find_sysfs_devs_nl_cb()?

Reply:
No, it's not expected to change.
On the other hand, I don't expect this function to get called more than once/twice on init time.